### PR TITLE
First round of old storage stack deletion.

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: src/planning/plan/tests/planificator-test.ts [hunk 1]
```diff
@@ -41,11 +39,6 @@ describe('planificator', () => {
...
       assert(key.protocol.length > 0,
             `Invalid protocol in key for '${storageKeyBase}' planificator storage key base`);
-      if (!Flags.useNewStorageStack) {
-        const keyBase: KeyBase = key as KeyBase;
-        assert(keyBase.location && keyBase.location.length > 0,
-            `Invalid location in key for '${storageKeyBase}' planificator storage key base`);
-      }
     };
 
     verifySuggestion(storageKeyForTest(arc.id));
```
Comment (galganif) [inline src/planning/plan/tests/planificator-test.ts]: do you want to remove the lines below:
verifySuggestion('firebase://arcs-test.firebaseio.com/123-456-7890-abcdef/1_2_3');
 verifySuggestion('pouchdb://local/testdb/');

File: src/runtime/runtime.ts [hunk 2]
```diff
@@ -181,15 +181,13 @@ export class Runtime {
   // Note that this incorrectly assumes every storage key can be of the form `prefix` + `arcId`.
   // Should ids be provided to the Arc constructor, or should they be constructed by the Arc?
   // How best to provide default storage to an arc given whatever we decide?
-  newArc(name: string, storageKeyPrefix?: string | ((arcId: ArcId) => StorageKey) | null, options?: RuntimeArcOptions): Arc {
+  newArc(name: string, storageKeyPrefix?: ((arcId: ArcId) => StorageKey) | null, options?: RuntimeArcOptions): Arc {
```
Comment (galganif) [inline src/runtime/runtime.ts]: I think you can remove the '| null'

File: src/runtime/tests/arc-test.ts [hunk 3]
```diff
@@ -202,15 +197,15 @@ describe('Arc new storage', () => {
...
 });
 
-const doSetup = async () => Flags.useNewStorageStack ? setup(arcId => new VolatileStorageKey(arcId, '')) : setup('volatile://');
+const doSetup = async () => setup(arcId => new VolatileStorageKey(arcId, ''));
 
 describe('Arc', () => {
   it('idle can safely be called multiple times ', async () => {
     const runtime = Runtime.newForNodeTesting();
-    const arc = runtime.newArc('test', Flags.useNewStorageStack ? null : 'volatile://');
+    const arc = runtime.newArc('test', null);
```
Comment (galganif) [inline src/runtime/tests/arc-test.ts]: just runtime.newArc('test') (here and below)

File: src/runtime/tests/arc-test.ts [hunk 4]
```diff
@@ -808,10 +803,7 @@ describe('Arc', () => {
     //assert.strictEqual(slotComposer.slotsCreated, 1);
   });
 
-  it('serialization roundtrip preserves data for volatile stores', async function() {
-    if (Flags.useNewStorageStack) {
-      this.skip();
-    }
+  it.skip('serialization roundtrip preserves data for volatile stores', async () => {
```
Comment (galganif) [inline src/runtime/tests/arc-test.ts]: could you add a comment on why this is skipped?

File: src/runtime/tests/arc-test.ts [hunk 5]
```diff
@@ -1130,19 +1118,11 @@ describe('Arc', () => {
 
     arc2.dispose();
     assert.isEmpty(DriverFactory.providers);
-  }));
+  });
 });
 
 describe('Arc storage migration', () => {
   describe('when new storage enabled', () => {
```
Comment (galganif) [inline src/runtime/tests/arc-test.ts]: have only one describe, or even move to one of the existing one.

File: src/runtime/tests/particle-api-test.ts [hunk 6]
```diff
@@ -98,12 +98,11 @@ class ResultInspector {
 async function loadFilesIntoNewArc(fileMap: {[index:string]: string, manifest: string}): Promise<Arc> {
   const manifest = await Manifest.parse(fileMap.manifest);
   const runtime = new Runtime({loader: new Loader(null, fileMap), context: manifest});
-  return runtime.newArc('demo', Flags.useNewStorageStack ? null : 'volatile://');
+  return runtime.newArc('demo', null);
```
Comment (galganif) [inline src/runtime/tests/particle-api-test.ts]: return runtime.newArc('demo');

File: src/runtime/tests/particle-api-test.ts [hunk 7]
```diff
@@ -1317,7 +1316,7 @@ describe('particle-api', () => {
     // TODO(lindner): add strict rendering
     const slotComposer = new SlotComposer();
     const arc = new Arc({id: IdGenerator.newSession().newArcId('demo'),
-        storageKey: 'key', loader, slotComposer, context});
+        storageKey: null, loader, slotComposer, context});
```
Comment (galganif) [inline src/runtime/tests/particle-api-test.ts]: you can leave "storageKey: null" out

File: src/runtime/storageNG/storage-ng.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+export function singletonHandle<T extends CRDTReferenceableSingleton, U>(
+  store: ActiveStore<T>,
+  arc: Arc,
+  options?: HandleOptions
+): SingletonHandle<U> {
+  const type = options && options.type ? options.type : store.baseStore.type;
+  const storageKey = store.baseStore.storageKey.toString();
+  const ttl = options && options.ttl ? options.ttl : undefined;
+  const proxy = new StorageProxy<T>('argument no longer required', store, type, storageKey, ttl);
```
[the remark this answers is not in the document]
Comment (shans) [reply src/runtime/storageNG/storage-ng.ts]: Yeah, it does. Hmm; this should probably be something consistently plumbed up from the store as it's expected to look like an ID.

[the remark this answers is not in the document]
Comment (shans) [reply src/runtime/storageNG/storage-ng.ts]: store.baseStore.id seems to be appropriate. It's generated off the arc ID if not provided.

File: src/runtime/tests/arc-test.ts (hunk 4 above)
Comment (shans) [reply src/runtime/tests/arc-test.ts]: I don't actually know, sorry. It was skipped before so I kept it skipped in this patch.

**Pull request merged**